tests/unit/Infrastructure/DashboardBundle/Factory/EntityTypeFactoryTest.php

```diff
@@ -225,4 +226,140 @@ public function test_build_create_new_oauth20_rs_form()
+                }), \Mockery::on(function ($options) use ($expectation) {
+                    static::assertSame($expectation, $options);
+                    return true;
+                }))
+            ->once()
+            ->andReturn($this->form);
+
+        $form = $this->factory->createEditForm(
+            $manageEntity,
+            $this->service,
+            $environment,
+            $isCopy
+        );
+
+        $this->assertInstanceOf(FormType::class, $form);
+    }
+
+    public static function testOptions(): array
+    {
+        return [
+            [
+                false,
+                true,
+                Constants::STATE_PUBLICATION_REQUESTED,
+                Constants::ENVIRONMENT_PRODUCTION,
+                [
+                    'validation_groups' => [
+                        0 => 'Default',
+                        1 => 'production',
+                    ],
+                ]
+            ],
+            [
+                true,
+                true,
+                Constants::STATE_PUBLICATION_REQUESTED,
+                Constants::ENVIRONMENT_PRODUCTION,
+                [
+                    'validation_groups' => [
+                        0 => 'Default',
+                        1 => 'production',
+                    ],
+                ]
+            ],
+            [
+                true,
+                false,
+                Constants::STATE_PUBLICATION_REQUESTED,
+                Constants::ENVIRONMENT_PRODUCTION,
+                [
+                    'validation_groups' => ['Default', 'production'],
+                    'publish_button_label' => 'entity.edit.label.change',
+                ]
+            ],
+            [
+                false,
+                false,
+                Constants::STATE_PUBLICATION_REQUESTED,
+                Constants::ENVIRONMENT_PRODUCTION,
+                [
+                    'validation_groups' => ['Default', 'production'],
+                    'publish_button_label' => 'entity.edit.label.change',
+                ]
+            ],
+            [
+                false,
+                false,
+                Constants::STATE_PUBLISHED,
+                Constants::ENVIRONMENT_PRODUCTION,
+                [
+                    'validation_groups' => ['Default', 'production'],
+                    'publish_button_label' => 'entity.edit.label.change',
+                ]
+            ],
+            [
+                true,
+                false,
+                Constants::STATE_PUBLISHED,
+                Constants::ENVIRONMENT_PRODUCTION,
+                [
+                    'validation_groups' => ['Default', 'production'],
+                    'publish_button_label' => 'entity.edit.label.change',
+                ]
+            ],
+            [
+                false,
+                false,
+                Constants::STATE_PUBLISHED,
+                Constants::ENVIRONMENT_TEST,
+                [
+                ]
+            ],
```

Comment:
Wouldn't this result in some kind of exception? 
Is this a relevant test avenue?